sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -120,7 +121,16 @@ case class HiveTableScanExec(
 
     HiveShim.appendReadColumns(hiveConf, neededColumnIDs, output.map(_.name))
 
-    val deserializer = tableDesc.getDeserializerClass.getConstructor().newInstance()
+    val currentState = SessionState.get()
+    val deserializer = if (currentState != null) {
+      val originClassLoader = currentState.getConf.getClassLoader
+      currentState.getConf.setClassLoader(sparkSession.sharedState.jarClassLoader)
```

Comment:
Could we set the  correct class loader when we constructing `SessionState`? Maybe it's here:
https://github.com/apache/spark/blob/128ea37bda3dbc1d6ba2af762c7453ab7605d430/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L189-L193

Reply:
We fixed `CliSessionState` class loader issue when constructing it by [SPARK-28840](https://issues.apache.org/jira/browse/SPARK-28840), so could we fix `SessionState` class loader issue when constructing `SessionState`?


Hive related code:
https://github.com/apache/hive/blob/rel/release-2.3.6/ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java#L399

Reply:
> We fixed `CliSessionState` class loader issue when constructing it by [SPARK-28840](https://issues.apache.org/jira/browse/SPARK-28840), so could we fix `SessionState` class loader issue when constructing `SessionState`?

It's better to solve this when construct HiveClientImpl's SessionState.
And we can handle it with same method like [SPARK-28840](https://issues.apache.org/jira/browse/SPARK-28840)

Reply:
Thank you @AngersZhuuuu

Reply:
Another question is why this issue only occurs in JDK 11?

Reply:
> Another question is why this issue only occurs in JDK 11?

Still work on this to find root cause. Will add to desc when find it.